src/hotspot/share/gc/g1/heapRegion.hpp

```diff
@@ -144,8 +144,15 @@ class HeapRegion : public CHeapObj<mtGC> {
   // This version synchronizes with other calls to par_allocate_impl().
   inline HeapWord* par_allocate_impl(size_t min_word_size, size_t desired_word_size, size_t* actual_word_size);
 
+  // Return the address of the beginning of the block that contains "addr".
+  // "q" is a block boundary that is <= "addr"; "n" is the address of the
+  // next block (or the end of the HeapRegion.)
+  inline HeapWord* forward_to_block_containing_addr(HeapWord* q, HeapWord* n,
+                                                    const void* addr) const;
```

Comment:
Not sure how helpful this comment is; it's not that the text is hard to comprehend but rather this method is not a proper (internal) API -- it's highly dependent on its caller. Therefore, I suggest inlining it to its sole caller.